Tasks/Common/dotNetCliWrapper.ts

```diff
@@ -57,10 +56,10 @@ export class DotNetCliWrapper {
         args.push('--disable-interactive')
         args.push('true')
 
-        return this.executeAsync(args, additionalArgs)
+        return await this.execute(args, additionalArgs)
     }
 
-    public async lambdaDeployAsync(
```

Comment:
The commands to Amazon.Lambda.Tools should look up the full path to the dotnet-lambda executable instead of relying on the dotnet CLI to resolve that path via the PATH environment variable. The problem is in a CI environment the tools directory might not be in the PATH for the service user of TFS.

Reply:
This is not required, dotnet cli should not use PATH because it's thing thing that installs it. It should know where its tools folder is, and if not the CLI is broken.

Reply:
I don't understand your point. We know the dotnet CLI uses the path. If we don't like that it uses that path that doesn't change the need for us to deal with the fact it uses the path.

Reply:
The python version does it in this way and it works fine: https://github.com/awslabs/aws-lambda-builders/blob/develop/aws_lambda_builders/workflows/dotnet_clipackage/actions.py

Reply:
Yes, but that is a bug in the python version that should be fixed. The .NET Core support is very new in SAM and probably hasn't been used much in a CI/CD environment yet. I coded that and I know it was my mistake to not add it there.

The PowerShell version does do this and I added it because users ran into this problem.
https://github.com/aws/aws-lambda-dotnet/blob/master/PowerShell/Module/Private/_DeploymentFunctions.ps1#L344

Reply:
This sounds like a bug in the .net cli then since it was just used to install and it knows where its tools are, is there an issue filed on their repo?

Reply:
I'm sure the .NET CLI team would argue against it being a bug. A .NET Core Global tools don't have to start with dotnet like our `dotnet lambda` command does. The executable can be named anything like `fake` for the F# build system. In that case when a user installs the fake, `dotnet tool install -g fake-cli`, and then executes the command `fake` from the command line the CLI isn't involved at all. So you can't say the CLI should resolve the path to fake.

There is also the feature of naming your executable `dotnet-lambda` and if you type the command `dotnet lambda` the CLI redirect the call to an executable called `dotnet-lambda`. We could have just as easily called the executable aws-lambda and then the CLI would not have been involved at all when we executed. Now we preferred using the name `dotnet-lambda` because it scopes this tool for doing .NET and Lambda. If we called something like aws-lambda there would probably be an expectation the tool does more then just .NET.

Reply:
With some more effort, I was able to get it to support this and not break our existing customers